### PR TITLE
Forum sample rework

### PR DESCRIPTION
**Pull request by letmaik**
- Move business logic out of controllers into services to allow unit testing / debugging outside CCF using `ccf-app` polyfills.
- Move demo scripts to separate folder away from tests.
- Remove `esm` dependency in favour of native ESM support in Node.js.
- Fix auth header in demo script `demo/submit-opinions.ts` which isn't tested in CI and missed a fix that was already applied to the unit tests earlier.
- Fix display of errors in app website.
- Exclude mathjs 9.3.1 due to a [packaging bug](https://github.com/josdejong/mathjs/issues/2169).
- Remove `selfsigned` dependency in favour of using `node-forge` directly.
- Update to latest `tsoa` version and remove redundant (empty) entry script.
- Some better error logging during debugging.
- Extended docs.

### CODE REVIEW

**Review by achamayou — APPROVED** [review]

**Comment by letmaik** [comment]
CI failure is unrelated.

File: doc/build_apps/js_app_tsoa.rst [hunk 1]
```diff
@@ -41,10 +41,13 @@ The sample app has the following folder layout:
     │   │   └── site.ts
     │   ├── models
     │   │   └── poll.ts
+    │   ├── services
```
Comment (beejones) [inline doc/build_apps/js_app_tsoa.rst:44]: Are CCF/enclave apps called services? If not, I would suggest to change services into e.g. enclaveApp. Services is confusing.

File: samples/apps/forum/README.md [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
 Start the sandbox:
 
 ```sh
 npm start
 ```
 
-Open your browser at https://127.0.0.1:8000/app/site
+(Use `VERBOSE=1 npm start` for verbose output)
 
-Generate opinions, user identities and submit:
```
Comment (beejones) [inline samples/apps/forum/README.md:19]: These edits are really a great addition

File: doc/build_apps/js_app_tsoa.rst (hunk 1 above)
Comment (letmaik) [reply doc/build_apps/js_app_tsoa.rst:44]: This term comes from tsoa's docs: https://tsoa-community.github.io/docs/getting-started.html#defining-our-first-model. I've seen this before outside tsoa, another term would be `*Manager` but I'd like to stick to tsoa's docs here. The whole npm app runs inside CCF, not just the services.

**Pull request merged**